sale_commission/models/settlement.py

```diff
@@ -154,25 +155,38 @@ def make_invoices(self, journal, product, date=False):
 
 class SettlementLine(models.Model):
     _name = "sale.commission.settlement.line"
-
-    settlement = fields.Many2one(
-        "sale.commission.settlement", readonly=True, ondelete="cascade",
-        required=True)
-    agent_line = fields.Many2many(
-        comodel_name='account.invoice.line.agent',
-        relation='settlement_agent_line_rel', column1='settlement_id',
-        column2='agent_line_id', required=True)
-    date = fields.Date(related="agent_line.invoice_date", store=True)
-    invoice_line = fields.Many2one(
-        comodel_name='account.invoice.line', store=True,
-        related='agent_line.invoice_line')
-    invoice = fields.Many2one(
-        comodel_name='account.invoice', store=True, string="Invoice",
-        related='invoice_line.invoice_id')
-    agent = fields.Many2one(
-        comodel_name="res.partner", readonly=True, related="agent_line.agent",
-        store=True)
-    settled_amount = fields.Float(
-        related="agent_line.amount", readonly=True, store=True)
-    commission = fields.Many2one(
-        comodel_name="sale.commission", related="agent_line.commission")
+    _auto = False
+
+    settlement = fields.Many2one("sale.commission.settlement",
+                                 readonly=True)
+    date = fields.Date(readonly=True)
+    invoice_line = fields.Many2one(comodel_name='account.invoice.line',
+                                   readonly=True)
+    invoice = fields.Many2one(comodel_name='account.invoice',
+                              string="Invoice", readonly=True)
+    agent = fields.Many2one(comodel_name="res.partner", readonly=True)
+    settled_amount = fields.Float(readonly=True)
+    commission = fields.Many2one(comodel_name="sale.commission",
+                                 readonly=True)
+
+    def init(self, cr):
+        tools.drop_view_if_exists(cr, self._table)
```

Comment:
But you're introducing SQL, so I prefer to avoid it. Related fields are there to give in the worst case (not stored), the same performance, but with a straight syntax and with more flexibility (not all packed in an SQL sentence).